test/utils.cpp

```diff
@@ -38,6 +38,13 @@ int64_t prime_number(int64_t i) {
   return p.at(i);
 }
 
+void maybeEmptyAllocator() {
+  static int calls = 0;
+  if (++calls % 100 == 0) {
+    c10::cuda::CUDACachingAllocator::emptyCache();
```

Comment:
The criteria here feels somewhat arbitrary.

Can we do it slightly better and check for device stats instead (check for the percentage of reserved_bytes to the total memory on current device). And it would also be better to do this at the *beginning* of a test other then tear down.

https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.h#L211